lib/config/validation.ts

```diff
@@ -672,3 +656,51 @@ export async function validateConfig(
   warnings.sort(sortAll);
   return { errors, warnings };
 }
+
+function validateRegexManagerFields(
+  regexManager: RegexManagerConfig,
+  currentPath: string
+): ValidationMessage[] {
+  const errors: ValidationMessage[] = [];
+
+  if (is.nonEmptyArray(regexManager.matchStrings)) {
+    let validRegex = false;
+    for (const matchString of regexManager.matchStrings) {
+      try {
+        regEx(matchString);
+        validRegex = true;
+      } catch (e) {
+        errors.push({
+          topic: 'Configuration Error',
+          message: `Invalid regExp for ${currentPath}: \`${String(
+            matchString
+          )}\``,
+        });
+      }
+    }
+    if (validRegex) {
+      const mandatoryFields = ['depName', 'currentValue', 'datasource'];
```

Comment:
I would always validate all mandatory fields, so users get all errors at once and don't need to run validation multiple times to get next error

Reply:
Doesn't the above return statement come in between that ?

Reply:
I think its okay to let these mandatory field errors not be shown if there is no matchStrings, since these mandatory fields come into play only if we are able to match against a string.